File: mobile/src/main/java/com/windscribe/mobile/welcome/fragment/FragmentCallback.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.welcome.fragment;

public interface FragmentCallback {

    void onAccountClaimButtonClick(String username, String password, String email, boolean ignoreEmptyEmail);

    void onBackButtonPressed();

    void onContinueWithOutAccountClick();

    void onForgotPasswordClick();

    void onLoginButtonClick(String username, String password, String twoFa);

    void onLoginClick();

    void onSignUpButtonClick(String username, String password, String email, boolean ignoreEmptyEmail);

    void onSkipToHomeClick();
}

File: mobile/src/main/java/com/windscribe/mobile/welcome/fragment/FragmentCallback.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.welcome.fragment

interface FragmentCallback {
    fun onAccountClaimButtonClick(
        username: String,
        password: String,
        email: String,
        ignoreEmptyEmail: Boolean,
        voucherCode: String
    )

    fun onBackButtonPressed()
    fun onContinueWithOutAccountClick()
    fun onForgotPasswordClick()
    fun onLoginButtonClick(username: String, password: String, twoFa: String)
    fun onLoginClick()
    fun onEmergencyClick()
    fun onSignUpButtonClick(
        username: String,
        password: String,
        email: String,
        referralUsername: String,
        ignoreEmptyEmail: Boolean,
        voucherCode: String
    )

    fun onSkipToHomeClick()
}